File: data/src/main/java/by/grodno/vasili/data/datasource/retrofit/UserApiService.java
package by.grodno.vasili.data.datasource.retrofit;

import java.util.List;

import by.grodno.vasili.data.response.OrganizationResponse;
import by.grodno.vasili.data.response.UserResponse;
import io.reactivex.Maybe;
import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface UserApiService {
    /**
     * Get all users
     * @param since The integer ID of the last User that you've seen
     * @return Observable list of {@link UserResponse}
     */
    @GET("users")
    Single<List<UserResponse>> getUsers(@Query("since") int since);

    /**
     * Get one user
     * @param username Username of user
     * @return Observable {@link UserResponse}
     */
    @GET("users/{username}")
    Maybe<UserResponse> getUser(@Path("username") String username);

    /**
     * Get organization for user
     * @param username Username of user
     * @return Single observable list of {@link OrganizationResponse}
     */
    @GET("users/{username}/orgs")
    Single<List<OrganizationResponse>> getOrganizations(@Path("username") String username);
}

File: data/src/main/java/by/grodno/vasili/data/datasource/retrofit/UserApiService.kt
package by.grodno.vasili.data.datasource.retrofit

import by.grodno.vasili.data.response.OrganizationResponse
import by.grodno.vasili.data.response.UserResponse
import io.reactivex.Maybe
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * Retrofit service for work with users.
 */
interface UserApiService {
    /**
     * Get all users.
     * [since] - The integer ID of the last User that you've seen.
     * @return Observable list of [UserResponse].
     */
    @GET("users")
    fun getUsers(@Query("since") since: Int): Single<List<UserResponse>>

    /**
     * Get one user
     * [username] Username of user.
     * @return Observable [UserResponse].
     */
    @GET("users/{username}")
    fun getUser(@Path("username") username: String): Maybe<UserResponse>

    /**
     * Get organization for user.
     * [username] Username of user.
     * @return Single observable list of [OrganizationResponse].
     */
    @GET("users/{username}/orgs")
    fun getOrganizations(@Path("username") username: String): Single<List<OrganizationResponse>>
}
